main.py

```diff
@@ -115,41 +118,29 @@ def setup_target_task_training(args, target_tasks, model, strict):
 
     if args.load_target_train_checkpoint not in ("none", ""):
         # This is to load a particular target train checkpoint.
-        checkpoint = glob.glob(args.load_target_train_checkpoint)
-        assert len(checkpoint) > 0, (
-            "Specified load_target_train_checkpoint not found: %r"
-            % args.load_target_train_checkpoint
-        )
-        assert len(checkpoint) == 1, (
-            "Too many checkpoints match pattern: %r" % args.load_target_train_checkpoint
-        )
-        best_path = checkpoint[0]
-        log.info("Loading existing model from %r...", best_path)
-        load_model_state(model, best_path, args.cuda, task_names_to_avoid_loading, strict=strict)
+        model_path = get_best_checkpoint_path(args, "target_train")
+        log.info("Loading existing model from %r...", model_path)
+        load_model_state(model, model_path, args.cuda, task_names_to_avoid_loading, strict=strict)
     else:
         if args.do_pretrain == 1:
-            best_pretrain = get_best_checkpoint_path(args.run_dir, "pretrain")
-            if best_pretrain:
+            model_path = get_best_checkpoint_path(args, "pretrain")
+            if model_path:
                 load_model_state(
-                    model, best_pretrain, args.cuda, task_names_to_avoid_loading, strict=strict
+                    model, model_path, args.cuda, task_names_to_avoid_loading, strict=strict
                 )
         else:
-            if args.transfer_paradigm == "finetune":
-                # We want to do target training without pretraining, thus
-                # we need to first create a checkpoint to come back to for each of
-                # the target tasks to finetune.
-                model_state = model.state_dict()
-                model_path = os.path.join(args.run_dir, "model_state_untrained_pre_target_train.th")
-                torch.save(model_state, model_path)
-            else:
-                # If do_pretrain = 0 and transfer_paradigm=frozen, then do_eval will evaluate on
-                # untrained encoder parameters.
-                assert_for_log(
-                    args.allow_untrained_encoder_parameters,
-                    "No best checkpoint found to evaluate. Set `allow_untrained_encoder_parameters` if you really want to use an untrained encoder.",
-                )
-                log.warning("Using untrained encoder parameters!")
-    return task_names_to_avoid_loading
+            # We want to do target training without pretraining, thus
+            # we need to first create a checkpoint to come back to for each of
+            # the target tasks to finetune.
+            model_state = model.state_dict()
+            assert_for_log(
+                args.allow_untrained_encoder_parameters,
+                "No best checkpoint found to evaluate. Set `allow_untrained_encoder_parameters` if you really want to use an untrained encoder.",
+            )
```

Comment:
Won't this fire now even if we're fine-tuning BERT? This was under a conditional before.

Reply:
If we're finetuning BERT and we don't have pretraining, wouldn't we want to make sure the user is extra  aware of that by switching allow_untrained_encoder_parameters to equal 1?

Reply:
I don't think so. I expect that >50% of the usage of jiant will just be finetuning BERT, or training and evaluating PRPN, or other setups without a pretraining phase.

Reply:
If we add a warning, let's add it only in the case where there are pretraining tasks specified, but do_pretrain isn't set, and there are no pretraining checkpoints. For the use cases I described, people would leave pretrain_tasks blank.